docs/docs/examples/guides/ur-harware-interface.md

```diff
@@ -0,0 +1,577 @@
+the Freedrive function of the teach pendant, which is, however, in some senses limited. It demands users to press on the
+button behind the pendant and then adjust the configuration almost joint by joint, with the feedback being quite light
+and unresponsive at times.
+
+For that purpose, AICA offers a hand guiding controller as a part of the UR hardware collection, based on UR's native
+force mode, and enriched with additional functionality such as spatial limits. To use it, simply click on the **+** icon
+in the **Controllers** list, and select the **UR Hand Guiding Controller**.
+
+<div class="text--center">
+  <img src={urHWIHandGuidingGraph} alt="Hand guiding controller graph" />
+</div>
+
+Since the controller uses force mode, it needs the name and the reference frame of the force/torque (FT) sensor. In the
+settings of the controller set the name of the sensor as **ur_tcp_fts_sensor** and the frame as **ur_tool0**.
+
+:::note
+
+Attempts to guide the robot by pushing on individual links will fail, as the forces must act on the FT sensor on the end
+effector.
+
+:::
+
+After pressing **Play**, the manipulator can be hand guided to points in space, driven by the forces applied on its end
+effector. In other words, it reads forces in the FT sensor, and "admits" them, trying to set the measured force to zero.
+The controller can be further tuned and adjusted by using its parameters:
+
+- Velocity/Force limits: the velocities and forces that can be applied by the controller in force mode (X, Y, Z, RZ, RY,
+  RZ).
+- Force/Torque threshold: the thresholds above which the hand guiding behavior is activated.
+- Compliant axes selection: the axes along which the robot can be hand guided (1-enabled, 0-disabled).
+- Hold delay: forces below the thresholds above for this duration will disable hand guiding.
+- X/Y/Z limits: spatial boundaries for the end effector motion, vectors of two values for the lower and upper limit.
+- Reference orientation: quaternion representing a desired orientation in base frame
+- Angular limit: allowed deviation from the reference orientation
+- Linear/Angular boundary strength: gains to apply restitution forces/torques, in case linear or angular limits are
+  exceeded.
+
+<div class="text--center">
+  <img src={urHWIHandGuidingParams} alt="Hand guiding controller parameters" style={{ width: '40%' }} />
+</div>
+
+## Impedance controller
+
+AICA's tailored impedance controller for UR robots takes advantage of UR's force mode to drive the robot in a compliant manner, 
+enabling safe, adaptive interaction with the environment.
+
+The controller takes a Cartesian state as input, and based on the error with respect to the current state, as well as the defined
+stiffness and damping parameters, tries to apply a wrench in space. In cases of free space, this will result in uninterrupted motion,
+while the robot will react compliantly to disturbances and forces applied on the end effector (where the force sensor lies).   
```

Comment:
```suggestion
The controller takes a desired Cartesian state as input. Then, it computes the error between input and current state. Using the defined
stiffness and damping parameters, it applies a desired wrench in space. In cases of obstacle-free space this will result in uninterrupted motion,
while the robot will react compliantly to disturbances and forces applied on the end effector (where the force sensor lies).   
```

I think this needs to be rephrased a bit (a bit too long in terms of sentences and can be a bit more precise). I added a suggestion just to highlight where I think we should rephrase, but up to you how this is done at the end